src/transformers/models/bloom/modeling_bloom.py

```diff
@@ -296,8 +293,9 @@ def forward(self, input, mask, max_positions):
             mask = torch.ones(input.shape[0], max_positions, dtype=torch.bool, device=input.device)
 
         mask = mask.to(input.device)
+        seq_ids = torch.arange(max_positions, device=input.device)
         causal_mask = (
-            torch.tril(torch.ones((max_positions, max_positions), dtype=torch.bool))
+            (seq_ids[None, None, :] <= seq_ids[None, :, None])
```

Comment:
Dont know if this is relevant but the original implementation outputs a tensor of rank 2, and your change outputs a tensor of rank 3. Should not be a big deal since we do reshape it afterwards but just wanted to point this out.

Reply:
Do we keep it like that?

Reply:
Fixed :) Thank you for the notice 🤗